File: EXAMPLES_Zig/jetgpio_example.zig
// Usage example of the JETGPIO library
// Compile with: zig build-exe jetgpio_example.zig -ljetgpio -lc
// Execute with: sudo ./jetgpio_example

const std = @import("std");
const jetgpio = @cImport({
    @cInclude("jetgpio.h");
});

pub fn main() !void {
    const init = jetgpio.gpioInitialise();

    if (init < 0) {
        std.debug.print("Error initiating jetgpio\n", .{});
    }
    // Setting up pin 3 as output
    const set1 = jetgpio.gpioSetMode(3, jetgpio.JET_OUTPUT);
    if (set1 < 0) {
        std.debug.print("Error setting pin 3\n", .{});
    }

    // Setting up pin 7 as input
    const set2 = jetgpio.gpioSetMode(7, jetgpio.JET_INPUT);
    if (set2 < 0) {
        std.debug.print("Error setting pin 7\n", .{});
    }

    // Writing 1 and 0 to pin 3 a 1 second intervals while reading pin 7
    var i: u8 = 0;
    var level: i32 = 0;
    var ret: i32 = 0;
    while (i < 5) {
        ret = jetgpio.gpioWrite(3, 1);
        std.time.sleep(1000);
        level = jetgpio.gpioRead(7);
        std.debug.print("level: {}\n", .{level});
        std.time.sleep(1000000000);
        ret = jetgpio.gpioWrite(3, 0);
        std.time.sleep(1000);
        level = jetgpio.gpioRead(7);
        std.debug.print("level: {}\n", .{level});
        std.time.sleep(1000000000);
        i += 1;
    }
    jetgpio.gpioTerminate();
}

File: github_bad/EXAMPLES_C++/jetgpio_example.cpp
/* Usage example of the JETGPIO library
 * Compile with: g++ -Wall -o jetgpio_example jetgpio_example.cpp -ljetgpio
 * Execute with: sudo ./jetgpio_example
 */

#include <iostream>
#include <unistd.h>
#include <jetgpio.h>

int main(int argc, char *argv[])
{
  int Init;

  Init = gpioInitialise();
  if (Init < 0)
    {
      /* jetgpio initialisation failed */
      printf("Jetgpio initialisation failed. Error code:  %d\n", Init);
      exit(Init);
    }
  else
    {
      /* jetgpio initialised okay*/
      printf("Jetgpio initialisation OK. Return code:  %d\n", Init);
    }	

  // Setting up pin 3 as OUTPUT and 7 as INPUT
  int stat1 = gpioSetMode(3, JET_OUTPUT);
  if (stat1 < 0)
    {
      /* gpio setting up failed */
      printf("gpio setting up failed. Error code:  %d\n", stat1);
      exit(Init);
    }
  else
    {
      /* gpio setting up okay*/
      printf("gpio setting up okay. Return code:  %d\n", stat1);
    }

  int stat2 = gpioSetMode(7, JET_INPUT);
  if (stat2 < 0)
    {
      /* gpio setting up failed */
      printf("gpio setting up failed. Error code:  %d\n", stat2);
      exit(Init);
    }
  else
    {
      /* gpio setting up okay*/
      printf("gpio setting up okay. Return code:  %d\n", stat2);
    }

  // Writing 1 and 0 to pin 3 at 1 second intervals while reading pin 7
  int x =0;
  int level = 0;
  while (x<5) {
    gpioWrite(3, 1);
    usleep(1000);
    level = gpioRead(7);
    printf("level: %d\n",level);
    sleep(1);
    gpioWrite(3, 0);
    usleep(1000);
    level = gpioRead(7);
    printf("level: %d\n",level);
    sleep(1);
    x++;
  }

  // Terminating library 
  gpioTerminate();

  exit(0);
	
}

